cli/cmd/upgrade.go

```diff
@@ -251,6 +251,13 @@ func upgrade(ctx context.Context, k *k8s.KubernetesAPI, flags []flag.Flag, stage
 			return bytes.Buffer{}, err
```

Comment:
I suspect that in the Helm case, `loadStoredValuesLegacy` probably returns an error because it tries to load the legacy Linkerd config which is empty.  This results in the PEM encoding error message because the CA cert field is empty and thus cannot be decoded.

One possible approach would be to modify `loadStoredValuesLegacy` to return `nil, nil` if the legacy Linkerd config is empty.  This lets us differentiate the following situations:

1. linkerd-config-overrides exists
  * recent Linkerd CLI was used to install
  * `loadStoredValues` returns non-nil
  * we can use those values as normal
2. linkerd-config-overrides does not exist but legacy config does
  * legacy Linkerd CLI was used to install
  * `loadStoredValuesLegacy` returns non-nil
  *  we can use those values as normal
 3. linkerd-config-overrides and legacy config both do not exist
   * Helm was used to install or linkerd-config-overrides has been deleted
   * We should return an error message
   * Note that once https://github.com/linkerd/linkerd2/pull/5636 merges, we can update this error message suggesting that if Helm was not used to install, the linkerd-config-overrides secret will need to be repaired.  (But we don't need to worry about that in this PR)